jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt

```diff
@@ -108,11 +114,39 @@ class DefaultCodeWhispererFileContextProvider(private val project: Project) : Fi
     override suspend fun extractSupplementalFileContext(psiFile: PsiFile, targetContext: FileContextInfo): SupplementalContextInfo? {
         val startFetchingTimestamp = System.currentTimeMillis()
         val isTst = isTestFile(psiFile)
+        val userGroup = CodeWhispererUserGroupSettings.getInstance().getUserGroup()
+        val language = targetContext.programmingLanguage
 
         val chunks = if (isTst && targetContext.programmingLanguage.isUTGSupported()) {
-            extractSupplementalFileContextForTst(psiFile, targetContext)
+            if (userGroup == CodeWhispererUserGroup.CrossFile) {
+                extractSupplementalFileContextForTst(psiFile, targetContext)
+            } else {
+                emptyList()
+            }
         } else if (!isTst && targetContext.programmingLanguage.isSupplementalContextSupported()) {
-            extractSupplementalFileContextForSrc(psiFile, targetContext)
+            when (language) {
+                is CodeWhispererJava -> extractSupplementalFileContextForSrc(psiFile, targetContext)
+
+                is CodeWhispererPython -> if (userGroup == CodeWhispererUserGroup.CrossFile) {
+                    extractSupplementalFileContextForSrc(psiFile, targetContext)
+                } else {
+                    emptyList()
+                }
+
+                is CodeWhispererJavaScript -> if (userGroup == CodeWhispererUserGroup.CrossFile) {
+                    extractSupplementalFileContextForSrc(psiFile, targetContext)
+                } else {
+                    emptyList()
+                }
+
+                is CodeWhispererTypeScript -> if (userGroup == CodeWhispererUserGroup.CrossFile) {
+                    extractSupplementalFileContextForSrc(psiFile, targetContext)
+                } else {
+                    emptyList()
+                }
```

Comment:
Can these be combined into one is block using 'or'

Reply:
sg